plugin/nfs/nfs.py

```diff
@@ -0,0 +1,571 @@
+
+    def exports(self, search_key=None, search_value=None, flags=0):
+        """List existing exports"""
+        exports = NFSPlugin._load_exports()
+        return search_property(exports, search_key, search_value)
+
+    def export_fs(self, fs_id, export_path, root_list, rw_list, ro_list,
+                  anon_uid=NfsExport.ANON_UID_GID_NA,
+                  anon_gid=NfsExport.ANON_UID_GID_NA,
+                  auth_type=None, options=None, flags=None):
+        """Add an export"""
+
+        if fs_id is None and export_path is None:
+            raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                           'Must provide fs_id or export_path')
+
+        if fs_id is None:
+            fs_id = get_fsid(export_path)
+            if fs_id is None:
+                raise LsmError(ErrorNumber.NOT_FOUND_FS,
+                               'FileSystem not found')
+
+        if export_path is None:
+            export_path = NFSPlugin._get_fsid_path(fs_id)
+            if export_path is None:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                               'Could not locate filesystem')
+
+        if auth_type is not None:
+            authlist = NFSPlugin._AUTH_LIST
+            if auth_type not in authlist:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                               'Unsupported auth type')
+
+        for host in root_list:
+            if host not in rw_list and host not in ro_list:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT,
+                               'Root hosts must also be in rw or ro lists')
+
+        expid = NFSPlugin._export_id(export_path, auth_type, anon_uid,
+                                     anon_gid, options)
+
+        newexp = NfsExport(expid, fs_id, export_path,
+                           auth_type, root_list, rw_list, ro_list,
+                           anon_uid, anon_gid, options)
+
+        efile = NFSPlugin._open_exports(readonly=False)
+        exports = NFSPlugin._read_exports(efile)
+        result = None
+
+        for idx, oldexp in enumerate(exports):
+            if NFSPlugin._match_path(newexp, oldexp):
+                exports[idx] = newexp
+                result = exports[idx]
+                newexp = None
+        if newexp is not None:
+            exports.append(newexp)
+            result = newexp
+
+        NFSPlugin._write_exports(efile, exports)
+        NFSPlugin._close_exports(efile)
+        NFSPlugin._update_exports()
+        return result
+
+    def export_remove(self, export, flags=0):
+        try:
+            efile = NFSPlugin._open_exports(readonly=False)
+            exports = NFSPlugin._read_exports(efile)
+            filtered = NFSPlugin._filter_export_byid(exports, export.id)
+            NFSPlugin._write_exports(efile, filtered)
+            NFSPlugin._close_exports(efile)
+            NFSPlugin._update_exports()
+        except Exception:
+            raise
```

Comment:
There is no need to do exception handling if we just raise whatever exception we got.